source/common/router/router.h

```diff
@@ -198,7 +198,12 @@ class Filter : Logger::Loggable<Logger::Id::router>,
     return callbacks_->connection();
   }
   const Http::HeaderMap* downstreamHeaders() const override { return downstream_headers_; }
-
+  absl::optional<std::function<bool(uint32_t, const Upstream::Host&)>>
```

Comment:
Also, for this function in particular, the `absl::optional` isn't explicitly needed as an std::function can just be nullptr. I don't really care one way or the other but just pointing that out.